src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +284,165 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(
+        self, run_request: RunTaskRequest, task_execution: _TaskExecution
+    ):
+        response = RunTaskResponse(
+            task_id=run_request.task_id,
+            outcome_code=TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE,
+            failure_reason=TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR,
+            stderr="",
+            stdout="",
+        )
+
+        try:
+
+            def progress_reporter(progress: Progress):
```

Comment:
`def progress_reporter` can be moved outside of the try block, making try block smaller makes it much more clear what is it for (RunTaskHandler)

Reply:
Done